table_test.go

```diff
@@ -579,94 +581,110 @@ func Test_Table_Concurrency(t *testing.T) {
-//					{Name: "label1", Value: id},
-//					{Name: "label2", Value: "value2"},
-//				},
-//				Stacktrace: []uuid.UUID{
-//					{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1},
-//					{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x2},
-//				},
-//				Timestamp: rand.Int63(),
-//				Value:     int64(i),
-//			})
-//		}
-//
-//		buf, err := rows.ToBuffer(config.schema)
-//		require.NoError(b, err)
-//
-//		buf.Sort()
-//		return buf
-//	}
-//
-//	// Pre-generate all rows we're inserting
-//	inserts := make(map[string]*dynparquet.Buffer, writers)
-//	for i := 0; i < writers; i++ {
-//		id := uuid.New().String()
-//		inserts[id] = generateRows(id, rows*iterations)
-//	}
-//
-//	b.ResetTimer()
-//
-//	for i := 0; i < b.N; i++ {
-//
-//		// Create table for test
-//		table := db.Table(uuid.New().String(), config, log.NewNopLogger())
-//
-//		// Spawn n workers that will insert values into the table
-//		wg := &sync.WaitGroup{}
-//		for id := range inserts {
-//			wg.Add(1)
-//			go func(id string, tbl *Table, w *sync.WaitGroup) {
-//				defer w.Done()
-//				for i := 0; i < iterations; i++ {
-//					if err := tbl.Insert(inserts[id][i*rows : i*rows+rows]); err != nil {
-//						fmt.Println("Received error on insert: ", err)
-//					}
-//				}
-//			}(id, table, wg)
-//		}
-//		wg.Wait()
-//
-//		b.StopTimer()
-//
-//		// Wait for all compaction routines to complete
-//		table.Sync()
-//
-//		// Calculate the number of entries in database
-//		totalrows := int64(0)
-//		err := table.Iterator(memory.NewGoAllocator(), func(ar arrow.Record) error {
-//			totalrows += ar.NumRows()
-//			defer ar.Release()
-//
-//			return nil
-//		})
-//		require.NoError(b, err)
-//		require.Equal(b, int64(rows*iterations*writers), totalrows)
-//
-//		b.StartTimer()
-//	}
-//}
+func Benchmark_Table_Insert_10Rows_10Iter_10Writers(b *testing.B) {
+	benchmarkTableInserts(b, 10, 10, 10)
+}
+
+func Benchmark_Table_Insert_100Row_100Iter_100Writers(b *testing.B) {
+	benchmarkTableInserts(b, 100, 100, 100)
+}
+
+func benchmarkTableInserts(b *testing.B, rows, iterations, writers int) {
+	var (
+		ctx    = context.Background()
+		config = NewTableConfig(
+			dynparquet.NewSampleSchema(),
+		)
+		c = New(prometheus.NewRegistry(), 512, 512*1024*1024)
+	)
+	db, err := c.DB("test")
+	require.NoError(b, err)
+	generateRows := func(id string, n int) *dynparquet.Buffer {
+		rows := make(dynparquet.Samples, 0, n)
+		for i := 0; i < n; i++ {
+			rows = append(rows, dynparquet.Sample{
+				Labels: []dynparquet.Label{ // TODO would be nice to not have all the same column
+					{Name: "label1", Value: id},
+					{Name: "label2", Value: "value2"},
+				},
+				Stacktrace: []uuid.UUID{
+					{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1},
+					{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x2},
+				},
+				Timestamp: rand.Int63(),
```

Comment:
Probably fine for a start, but this isn't really representative of our typical workloads, inserts don't happen with random, but roughly monotonically increasing timestamps. Probably good to optimize these cases as well, but we'll probably see CPU time spent in things that a real workload we use FrostDB for doesn't.